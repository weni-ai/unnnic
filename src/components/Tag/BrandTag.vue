<template>
  <div
    :class="{
      'unnnic-brand-tag': true,
      'unnnic-brand-tag--disabled': !disabled,
      'unnnic-brand-tag--clickable': clickable,
    }"
  >
    <span class="unnnic-brand-tag__label">{{ text }}</span>
    <unnnic-icon-svg
      :icon="disabled ? 'close-1' : 'add-1'"
      :scheme="disabled ? 'neutral-snow': 'brand-weni-dark'"
      class="unnnic-brand-tag__icon"
      size="xs"
    />
  </div>
</template>

<script>
import UnnnicIconSvg from '../Icon.vue';

export default {
  name: 'unnnic-brand-tag',
  components: {
    UnnnicIconSvg,
  },
  props: {
    text: {
      type: String,
      default: null,
    },
    clickable: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    icon: {
      type: String,
      default: null,
    },
  },
  methods: {
    closeClicked() {
      if (!this.closeClicked) return;
      this.$emit('close-click');
    },
  },
};
</script>

<style lang="scss" scoped>
@import "../../assets/scss/unnnic.scss";

.unnnic-brand-tag {
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: $unnnic-border-radius-pill;
  border: 1px solid $unnnic-color-brand-weni-soft;
  height: 1.875 * $unnnic-font-size;
  white-space: nowrap;
  transition: 0.2s;

  color: $unnnic-color-neutral-snow;
  background-color: $unnnic-color-brand-weni-dark;
  &:hover {
    background-color: $unnnic-color-brand-weni-soft;
  }

  &--disabled {
    color: $unnnic-color-brand-weni-dark;
    background-color: $unnnic-color-background-snow;
    &:hover {
      background-color: rgba($unnnic-color-brand-weni, $unnnic-opacity-level-extra-light);
    }
  }
  &--clickable {
    cursor: pointer;
  }

  &__label {
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-md;
    font-weight: $unnnic-font-weight-regular;
    line-height: ($unnnic-font-size-body-md + $unnnic-line-height-medium);
    padding: $unnnic-spacing-stack-nano $unnnic-inline-ant;
  }

  &__icon {
    margin-right: $unnnic-inline-ant;
    cursor: pointer;
  }
}
</style>
