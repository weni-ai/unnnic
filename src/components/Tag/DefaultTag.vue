<template>
  <div
    :class="`unnnic-tag
        ${!disabled ? `unnnic-tag-scheme--${scheme}` : `unnnic-tag--disabled`}
        ${clickable ? 'unnnic-tag--clickable' : ''}`"
  >
    <span
      :class="`unnnic-tag__label
      ${hasCloseIcon ? 'unnnic-tag__label--hasIcon' : ''}
      ${disabled ? 'unnnic-tag__label--disabled' : ''}`"
      >{{ text }}</span
    >
    <unnnic-icon-svg
      v-if="hasCloseIcon"
      icon="close-1"
      class="unnnic-tag__icon"
      scheme="neutral-darkest"
      size="xs"
      @click.stop="$emit('close')"
    />
  </div>
</template>

<script>
import UnnnicIconSvg from '../Icon.vue';

export default {
  name: 'unnnic-tag',
  components: {
    UnnnicIconSvg,
  },
  props: {
    text: {
      type: String,
      default: null,
    },
    clickable: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    hasCloseIcon: {
      type: Boolean,
      default: false,
    },
    scheme: {
      type: String,
      default: 'aux-purple',
    },
  },
  methods: {
    closeClicked() {
      if (!this.closeClicked) return;
      this.$emit('close-click');
    },
  },
};
</script>

<style lang="scss" scoped>
@import '../../assets/scss/unnnic.scss';

@each $name, $color in $scheme-colors {
  .unnnic-tag-scheme--#{$name} {
    background-color: rgba($color, $unnnic-opacity-level-light);
    outline-style: solid;
    outline-color: rgba($color, $unnnic-opacity-level-extra-light);
    outline-width: $unnnic-border-width-thinner;
    outline-offset: -$unnnic-border-width-thinner;
  }
}

.unnnic-tag {
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: $unnnic-border-radius-pill;

  &--disabled {
    background-color: $unnnic-color-background-sky;
  }

  &--clickable {
    cursor: pointer;

    @each $name, $color in $scheme-colors {
      &.unnnic-tag-scheme--#{$name}:hover {
        outline-style: solid;
        outline-color: $unnnic-color-neutral-cleanest;
        outline-width: $unnnic-border-width-thinner;
        outline-offset: -$unnnic-border-width-thinner;
      }
    }
  }

  &__label {
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-md;
    font-weight: $unnnic-font-weight-regular;
    line-height: ($unnnic-font-size-body-md + $unnnic-line-height-medium);
    padding: $unnnic-spacing-stack-nano $unnnic-inline-ant;
    color: $unnnic-color-neutral-darkest;

    &--disabled {
      color: $unnnic-color-neutral-cloudy;
    }

    &--hasIcon {
      padding-left: $unnnic-inline-ant;
      padding-right: $unnnic-spacing-stack-nano;
    }
  }

  &__icon {
    margin-right: $unnnic-inline-ant;
    cursor: pointer;
  }
}
</style>
