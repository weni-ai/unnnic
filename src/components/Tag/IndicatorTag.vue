<template>
  <div
    :class="`unnnic-tag
        ${clickable ? 'unnnic-tag--clickable' : ''}`"
  >
    <tool-tip :enabled="enableTooltip" :text="tooltipText" side="bottom">
      <span class="unnnic-tag__count">{{ count }}</span>
    </tool-tip>
    <span class="unnnic-tag__label">{{ text }}</span>
    <unnnic-icon-svg
      v-if="hasBackButton"
      icon="arrow-right-1-1"
      class="unnnic-tag__icon"
      scheme="neutral-darkest"
      size="xs"
    />
  </div>
</template>

<script>
import UnnnicIconSvg from '../Icon.vue';
import ToolTip from '../ToolTip/ToolTip.vue';

export default {
  name: 'unnnic-tag',
  components: {
    UnnnicIconSvg,
    ToolTip,
  },
  props: {
    text: {
      type: String,
      default: null,
    },
    tooltipText: {
      type: String,
      default: null,
    },
    count: {
      type: Number,
      default: 0,
    },
    clickable: {
      type: Boolean,
      default: false,
    },
    hasBackButton: {
      type: Boolean,
      default: false,
    },
    enableTooltip: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss" scoped>
@import "../../assets/scss/unnnic.scss";

.unnnic-tag {
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: $unnnic-border-radius-sm;
  background-color: $unnnic-color-background-sky;
  padding: $unnnic-inset-nano;
  border: $unnnic-border-width-thinner solid $unnnic-color-background-sky;

  &:hover {
    border: $unnnic-border-width-thinner solid $unnnic-color-neutral-cleanest;
  }

  &--clickable {
    cursor: pointer;
  }

  &__count {
    background-color: $unnnic-color-neutral-soft;
    border-radius: $unnnic-border-radius-sm;
    padding: $unnnic-inline-nano;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-md;
    font-weight: $unnnic-font-weight-regular;
    line-height: ($unnnic-font-size-body-md + $unnnic-line-height-medium);
    color: $unnnic-color-neutral-darkest;

  }

  &__label {
    padding: 0 $unnnic-inline-xs;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-md;
    font-weight: $unnnic-font-weight-regular;
    line-height: ($unnnic-font-size-body-md + $unnnic-line-height-medium);
    color: $unnnic-color-neutral-darkest;
  }

  &__icon {
    margin-right: $unnnic-inline-ant;
  }
}
</style>
