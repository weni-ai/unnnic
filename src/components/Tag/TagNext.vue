<template>
  <section :class="['unnnic-tag-next', `unnnic-tag-next--${scheme}`]">
    <p class="unnnic-tag-next__text">
      {{ text }}
    </p>
  </section>
</template>

<script>
export default {
  name: 'TagNext',
  props: {
    text: {
      type: String,
      default: null,
    },
    scheme: {
      type: String,
      default: 'weni',
    },
  },
};
</script>

<style lang="scss" scoped>
@use '@/assets/scss/unnnic' as *;

.unnnic-tag-next {
  display: flex;
  align-items: center;
  justify-content: center;
  width: auto;

  border-radius: $unnnic-border-radius-pill;
  padding: 0 $unnnic-spacing-xs;

  font-family: $unnnic-font-family-secondary;
  font-size: $unnnic-font-size-body-md;
  font-weight: $unnnic-font-weight-regular;
  line-height: ($unnnic-font-size-body-md + $unnnic-line-height-medium);

  &__text {
    margin: 0;

    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  &--aux-purple {
    background-color: $unnnic-color-aux-purple-100;
    color: $unnnic-color-aux-purple-500;
  }

  &--weni {
    background-color: $unnnic-color-weni-100;
    color: $unnnic-color-weni-600;
  }
}
</style>
