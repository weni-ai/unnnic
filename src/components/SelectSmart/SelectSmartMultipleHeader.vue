<template>
  <div class="unnnic-select-smart__options__multiple">
    <div
      v-if="selectedOptions[0]"
      class="unnnic-select-smart__options__multiple__selecteds__container"
    >
      <div class="unnnic-select-smart__options__multiple__selecteds">
        <tag
          v-for="option in firstMultipleSelecteds"
          class="unnnic-select-smart__options__multiple__selecteds__option"
          :key="option.value"
          :text="option.label"
          hasCloseIcon
          @close="unselectOption(option)"
        />
        <p
          v-if="selectedOptions.length > multipleSelectedsTags"
          class="unnnic-select-smart__options__multiple__selecteds__remaining"
        >
          +{{ selectedOptions.length - multipleSelectedsTags }}
        </p>
      </div>
      <icon-svg
        class="unnnic-select-smart__options__multiple__selecteds__clear"
        icon="close-1"
        size="xs"
        clickable
        @click="clearSelectedOptions"
      />
    </div>
    <p v-if="!selectedOptions[0]" class="unnnic-select-smart__options__multiple--without-multiples">
      {{ withoutSelectsMessage || i18n('without_multiple_selected') }}
    </p>
  </div>
</template>

<script>
import Tag from '../Tag/Tag.vue';
import IconSvg from '../Icon.vue';
import UnnnicI18n from '../../mixins/i18n';

export default {
  name: 'UnnnicSelectSmartMultipleHeader',
  mixins: [UnnnicI18n],
  components: {
    Tag,
    IconSvg,
  },
  props: {
    selectedOptions: {
      type: Array,
      required: true,
    },
    withoutSelectsMessage: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      multipleSelectedsTags: 2,

      defaultTranslations: {
        without_multiple_selected: {
          'pt-br': 'Nenhuma opção selecionada',
          en: 'No option selected',
          es: 'Ninguna opción seleccionada',
        },
      },
    };
  },
  computed: {
    firstMultipleSelecteds() {
      const { selectedOptions, multipleSelectedsTags } = this;
      const selectedArray = [];

      for (let i = 0; i < multipleSelectedsTags; i += 1) {
        selectedArray.push(selectedOptions?.[i]);
      }

      return selectedArray.filter((option) => option !== undefined);
    },
  },
  methods: {
    clearSelectedOptions() {
      this.$emit('clear-selected-options');
    },
    unselectOption(option) {
      this.$emit('unselect-option', option);
    },
  },
};
</script>

<style lang="scss" scoped>
@import '../../assets/scss/unnnic.scss';

.unnnic-select-smart__options__multiple {
  border-bottom: 1px solid $unnnic-color-neutral-soft;

  &__selecteds {
    display: flex;
    align-items: center;

    color: $unnnic-color-neutral-dark;

    &__container {
      display: flex;
      justify-content: space-between;
      align-items: center;
    }

    &__option {
      margin: $unnnic-spacing-xs;
      margin-right: 0;

      &.unnnic-tag {
        outline-color: $unnnic-color-neutral-light;
        background-color: $unnnic-color-neutral-light;

        color: $unnnic-color-neutral-dark;
      }
    }

    &__remaining {
      margin-left: $unnnic-spacing-xs;

      line-height: $unnnic-font-size-body-md + $unnnic-line-height-small;
      font-size: $unnnic-font-size-body-gt;
    }

    &__clear {
      margin-right: $unnnic-spacing-sm;
    }
  }

  &--without-multiples {
    margin: 0;

    color: $unnnic-color-neutral-cleanest;
    line-height: $unnnic-font-size-body-md + $unnnic-line-height-medium;
    font-size: $unnnic-font-size-body-md;

    padding: $unnnic-spacing-ant;
  }
}
</style>
