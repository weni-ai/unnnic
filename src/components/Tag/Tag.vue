<template>
    <component
      class="unnnic-tag-content"
      :is="currentComponent"
      :v-bind="$attrs"
      :text="text"
      :disabled="disabled"
      :hasCloseIcon="hasCloseIcon"
      :scheme="scheme"
      :count="count"
      :hasBackButton="hasBackButton"
      :clickable="clickable"
      :tooltipText="tooltipText"
      :enableTooltip="enableTooltip"
      :icon="icon"
      @click="$emit('click')"
      @close="$emit('close')"
    />
</template>

<script>
import DefaultTag from './DefaultTag.vue';
import IndicatorTag from './IndicatorTag.vue';
import BrandTag from './BrandTag.vue';

export default {
  name: 'unnnic-tag',
  props: {
    type: {
      type: String,
      default: 'default',
      validator(value) {
        return (
          ['default', 'indicator', 'brand'].indexOf(value) !== -1
        );
      },
    },
    text: {
      type: String,
      default: null,
    },
    tooltipText: {
      type: String,
      default: null,
    },
    clickable: {
      type: Boolean,
      default: false,
    },
    count: {
      type: Number,
      default: 0,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    hasCloseIcon: {
      type: Boolean,
      default: false,
    },
    scheme: {
      type: String,
      default: 'aux-purple',
    },
    hasBackButton: {
      type: Boolean,
      default: false,
    },
    enableTooltip: {
      type: Boolean,
      default: false,
    },
    icon: {
      type: String,
      default: null,
    },
  },
  computed: {
    currentComponent() {
      if (this.type === 'indicator') return IndicatorTag;
      if (this.type === 'brand') return BrandTag;
      return DefaultTag;
    },
  },
};
</script>
