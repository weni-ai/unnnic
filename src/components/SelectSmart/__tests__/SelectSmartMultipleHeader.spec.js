import { shallowMount } from '@vue/test-utils';
import SelectSmartMultipleHeader from '../SelectSmartMultipleHeader.vue';
import Tag from '@/components/Tag/Tag.vue';

describe('SelectSmartMultipleHeader.vue', () => {
  let wrapper;

  const mountWrapper = (props = {}) => {
    wrapper = shallowMount(SelectSmartMultipleHeader, {
      propsData: {
        selectedOptions: [],
        ...props,
      },
      stubs: {
        Tag,
      },
    });
  };

  beforeEach(() => {
    mountWrapper();
  });

  describe('initial rendering', () => {
    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without any selected options by default', () => {
      const withoutSelectsMessage = wrapper.find(
        '[data-testid="without-selects-message"]',
      );
      expect(withoutSelectsMessage.exists()).toBe(true);
      expect(withoutSelectsMessage.text()).toBe('No option selected');
    });

    it('displays the correct "without selects" message when no options are selected', async () => {
      await wrapper.setProps({ withoutSelectsMessage: 'No items selected' });
      const withoutSelectsMessage = wrapper.find(
        '[data-testid="without-selects-message"]',
      );
      expect(withoutSelectsMessage.text()).toBe('No items selected');
    });
  });

  describe('selected options display', () => {
    beforeEach(() => {
      mountWrapper({
        selectedOptions: [
          { value: 1, label: 'Option 1' },
          { value: 2, label: 'Option 2' },
          { value: 3, label: 'Option 3' },
        ],
      });
    });

    it('renders tags for the first multiple selected options', () => {
      const tags = wrapper.findAllComponents(Tag);
      expect(tags.length).toBe(2);
    });

    it('displays remaining selected options count correctly', () => {
      const remainingCount = wrapper.find('[data-testid="remaining-count"]');
      expect(remainingCount.exists()).toBe(true);
      expect(remainingCount.text()).toBe('+1');
    });
  });

  describe('unselectOption method', () => {
    it('emits "unselect-option" event when a Tag close icon is clicked', async () => {
      mountWrapper({
        selectedOptions: [
          { value: 1, label: 'Option 1' },
          { value: 2, label: 'Option 2' },
        ],
      });

      const firstTag = wrapper.findComponent(Tag);
      await firstTag.vm.$emit('close');
      expect(wrapper.emitted('unselect-option')).toBeTruthy();
      expect(wrapper.emitted('unselect-option')[0]).toEqual([
        { value: 1, label: 'Option 1' },
      ]);
    });
  });
});
