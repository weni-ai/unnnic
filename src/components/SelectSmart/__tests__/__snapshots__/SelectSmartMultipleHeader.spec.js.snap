// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectSmartMultipleHeader.vue > initial rendering > matches the snapshot 1`] = `
"<div data-v-4950cf71="" class="unnnic-select-smart__options__multiple">
  <!--v-if-->
  <p data-v-4950cf71="" class="unnnic-select-smart__options__multiple--without-multiples" data-testid="without-selects-message">No option selected</p>
</div>"
`;
